apps/src/templates/JsDebugger.jsx

```diff
@@ -0,0 +1,175 @@
+/**
+ * A React component for our JavaScript debugger UI. Returns a connected component
+ * so this can only be used in cases where we have a redux store.
+ */
+
+var connect = require('react-redux').connect;
+
+var i18n = require('../locale');
+var commonStyles = require('../commonStyles');
+var ProtectedStatefulDiv = require('./ProtectedStatefulDiv');
+
+var styles = {
+  noPadding: {
+    padding: 0
+  }
+};
+
+/**
+ * The console for our debugger UI
+ */
+var DebugConsole = function (props) {
+  var classes = 'debug-console';
+  if (!props.debugButtons) {
+    classes += ' no-commands';
+  }
+  if (!props.debugWatch) {
+    classes += ' no-watch';
+  }
+
+  return (
+    <div id="debug-console" className={classes}>
+      <div id="debug-output" className="debug-output"/>
+      <span className="debug-input-prompt">
+        &gt;
+      </span>
+      <div contentEditable spellCheck="false" id="debug-input" className="debug-input"/>
+    </div>
+  );
+};
+
+/**
+ * Buttons for stepping through code.
+ */
+var DebugButtons = function () {
+  return (
+    <div id="debug-commands" className="debug-commands">
+      <div id="debug-buttons">
+        <span> </span>
+        <button id="pauseButton" className="debugger_button">
+          <img src="/blockly/media/1x1.gif" className="pause-btn icon21"/>
+          {i18n.pause()}
+        </button>
+        <span> </span>
+        <button id="continueButton" className="debugger_button">
+          <img src="/blockly/media/1x1.gif" className="continue-btn icon21"/>
+          {i18n.continue()}
+        </button>
+        <span> </span>
+        <button id="stepOverButton" className="debugger_button">
+          <img src="/blockly/media/1x1.gif" className="step-over-btn icon21"/>
+          {i18n.stepOver()}
+        </button>
+
```

Comment:
What's with all the extra `<span> </span>` in this markup?

* Why isn't there one between stepOver and stepOut? I assume because we expect a line-break there.
* Why _is_ there one before pauseButton and one between pauseButton and continueButton (which are never displayed at the same time)?
* Could you just use nonbreaking spaces for this?